sunpy/net/dataretriever/sources/goes.py

```diff
@@ -52,33 +93,75 @@ def _get_url_for_timerange(self, timerange, **kwargs):
         ----------
         timerange: sunpy.time.TimeRange
             time range for which data is to be downloaded.
-        satellite_number : int
-            GOES satellite number (default = 15)
-        data_type : string
-            Data type to return for the particular GOES satellite. Supported
-            types depend on the satellite number specified. (default = xrs_2s)
+        Physobs : string
+            'INTENSITY', 'PARTICLE_FLUX' and 'IRRADIANCE', Case sensitive
+        Instrument: string
+            Fixed argument, 'goes' Case insensitive.
         """
-        # find out which satellite and datatype to query from the query times
-        sat_num = self._get_goes_sat_num(timerange.start, timerange.end)
-        base_url = 'http://umbra.nascom.nasa.gov/goes/fits/'
-        total_days = (timerange.end - timerange.start).days + 1 
-        all_dates = timerange.split(total_days)
-        result = list()
-#       Earlier we assumed that an entire range (consequently all dates within the range) were assigned the
-#       same GOES sat number, which meant we supplied timerange.start and timerange.end to
-#       _get_goes_sat_num. But now, we iterate over all the dates individually and hence
-#       _get_goes_sat_num gets each single date as an argument (the timerange.start and timerange.end
-#       for a single day is the day itself (trivial) ). That is why the same date. Its iterating over all the dates
-#       indivually.
-        for day in all_dates:
-            regex = "{date:%Y}/go{sat:02d}"
-            if (day.end < parse_time('1999/01/15')):
-                regex += "{date:%y%m%d}.fits"
-            else:
-                regex += "{date:%Y%m%d}.fits"
-            url = base_url + regex.format(
-                    date=day.end, sat=self._get_goes_sat_num(day.end, day.end)[0])
-            result.append(url)
+        physobs = kwargs['physobs']
+        def download_particle_flux(time_range):
+            now = datetime.datetime.now()
+            real_time = ((now - time_range.end).total_seconds() <= 300)
+            result = list()
+            total_days = (time_range.end - time_range.start).days + 1
+            all_dates = time_range.split(total_days)
+            url_pattern = 'ftp://ftp.swpc.noaa.gov/pub/lists/particle/{date:%Y%m%d}_G{char}_part_5m.txt'
+            base_url = 'http://umbra.nascom.nasa.gov/goes/fits/'
+            for day in all_dates:
+                result.extend([url_pattern.format(date=day.end, char=ch) for ch in ('p', 's')])
+                regex = "{date:%Y}/go{sat:02d}"
+                if (now - day.end).days > 3:
+                    url = base_url + regex.format(date=day.end, sat=self._get_goes_sat_num(day.end, day.end)[0])
+                    result.append(url)
```

Comment:
This url does not provide particle flux but x-ray flux (aka irradiance), so move this and the time logic from the "removed bit" to the `download_irradiance` function below.  Also, make sure the removed tests are updated to follow this case.